File: src/com/dafttech/terra/game/world/interaction/WeaponType.java
package com.dafttech.terra.game.world.interaction;

public class WeaponType {
    Skill repeatable, main1, main2, sub1, sub2;
    boolean providesFullSet = false;
    boolean allowsSubWeapon = true;

    public Skill getSkillForWeaponSkillSlotID(int id) {
        switch (id) {
        case 0:
            return repeatable;
        case 1:
            return main1;
        case 2:
            return main2;
        case 3:
            return sub1;
        case 4:
            return sub2;
        default:
            return null;
        }
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/world/interaction/WeaponType.scala
package com.dafttech.terra.game.world.interaction

class WeaponType {
  private[interaction] val repeatable: Skill = null
  private[interaction] val main1: Skill = null
  private[interaction] val main2: Skill = null
  private[interaction] val sub1: Skill = null
  private[interaction] val sub2: Skill = null
  private[interaction] val providesFullSet: Boolean = false
  private[interaction] val allowsSubWeapon: Boolean = true

  def getSkillForWeaponSkillSlotID(id: Int): Skill = id match {
    case 0 => repeatable
    case 1 => main1
    case 2 => main2
    case 3 => sub1
    case 4 => sub2
  }
}